prime-router/src/main/kotlin/cli/helpers/HL7DiffHelper.kt

```diff
@@ -0,0 +1,334 @@
+                if (inputComponents.size != outputComponents.size) {
+                    return listOf(
+                        Hl7Diff(
+                            segmentIndex,
+                            "Difference in number of components.",
+                            "",
+                            fieldNumber,
+                            secondaryFieldNum,
+                            tertiaryFieldNumber,
+                            segmentType
+                        )
+                    )
+                } else if (inputExtraComponents.numComponents() != outputExtraComponents.numComponents()) {
+                    return listOf(
+                        Hl7Diff(
+                            segmentIndex,
+                            "Difference in number of extra components.",
+                            "",
+                            fieldNumber,
+                            secondaryFieldNum,
+                            tertiaryFieldNumber,
+                            segmentType
+                        )
+                    )
+                } else {
+                    val compositeDifferences = mutableListOf<Hl7Diff>()
+                    inputComponents.zip(outputComponents).forEachIndexed { index, (i, o) ->
+                        val tertiaryFieldNumber2 = if (secondaryFieldNum == null) {
+                            null
+                        } else {
+                            index + 1
+                        }
+                        compositeDifferences.addAll(
+                            compareHl7Type(
+                                segmentIndex,
+                                i,
+                                o,
+                                segmentType,
+                                fieldNumber,
+                                secondaryFieldNum ?: (index + 1),
+                                tertiaryFieldNumber2
+                            )
+                        )
+                    }
+                    return compositeDifferences
+                }
+            }
+
+            input.javaClass != output.javaClass -> {
+                return listOf(
+                    Hl7Diff(
+                        segmentIndex,
+                        "Difference in type of field, ${input.javaClass}, ${output.javaClass}.",
+                        "",
+                        fieldNumber,
+                        secondaryFieldNum,
+                        tertiaryFieldNumber,
+                        segmentType
+                    )
+                )
+            }
+
+            else -> {
+                return listOf()
+            }
+        }
+    }
+
+    /**
+     * Map the pieces of the structure to their index and name
+     * ex.
+     * ...
+     * OBR|
+     * OCR|
+     * OBX|
+     * SPM|
+     * OBX|
+     * OBR|
+     * OBX|
+     * The last OBX would be indexed as 2-1-1, 2 because it's in the second observation_result
```

Comment:
Update this comment if you made the change per Michael's comment below?

Reply:
Actually reading this again its probably fine as is?

Reply:
Yep, still accurate.